packages/wxr/src/1.3/index.js

```diff
@@ -0,0 +1,665 @@
+
+		const decoder = new TextDecoder();
+
+		const writableStream = new WritableStream( {
+			write: ( chunk ) =>
+				new Promise( ( resolve ) => {
+					buffer += decoder.decode( chunk );
+					resolve();
+				} ),
+		} );
+
+		await this.stream( writableStream );
+
+		return buffer;
+	}
+
+	/**
+	 * Given a WritableStream, generate the WXR file, and write it to that stream.
+	 *
+	 * @param {WritableStream} writableStream The stream to write to.
+	 */
+	async stream( writableStream ) {
+		const writer = writableStream.getWriter();
+
+		await writer.ready;
+
+		await this.writeHeader( writer );
+
+		let tabs = 1;
+
+		// We want to process each part of the schema in sequence. However, since the
+		// processing function is asynchronous, we can't just use forEach(), as that will
+		// move on as soon as the processing function is called, rather than waiting for it
+		// to finish.
+		//
+		// Instead, we make use of reduce()'s behaviour, where it passes the result of the previous
+		// function call to the next function. As the previous function call is asynchronous, we can
+		// just wait for it to finish before proceeding.
+		await Object.entries( schema ).reduce(
+			async ( lock, [ store, storeDef ] ) => {
+				// Wait for the previous store processing to finish.
+				await lock;
+				await writer.ready;
+
+				// We'll need to at least lock the current data store.
+				const txStores = [ store ];
+
+				if ( store === 'comments' ) {
+					// Comments are handled inside individual posts, we can skip them.
+					return;
+				} else if ( store === 'posts' ) {
+					// We're in posts, so we need to lock the comments data store, too.
+					txStores.push( 'comments' );
+				}
+
+				// Start a transaction against the database.
+				const tx = this.db.transaction( txStores );
+
+				// Loop over every item in the current data store.
+				for await ( const cursor of tx.objectStore( store ) ) {
+					const datum = cursor.value;
+
+					if ( store === 'objects' ) {
+						const objectType = datum.type;
+
+						if ( objectType === 'contact-form' ) {
+							const output = [];
+
+							output.push( { email: datum.data.email } );
+							for ( const field of datum.data.fields ) {
+								const outputObject = {
+									_name: 'field',
+									_attrs: {
+										...field,
+									},
+								};
+
+								if ( field.type === 'select' ) {
+									delete outputObject._attrs.options;
+
+									outputObject._content = [];
+									for ( const selectOption of field.options ) {
+										outputObject._content.push( {
+											_name: 'option',
+											_attrs: {
+												value: selectOption.value,
+											},
+											_content: selectOption.text,
+										} );
+									}
+								}
+
+								output.push( outputObject );
+							}
+
+							this.write(
+								writer,
+								toXML(
+									{
+										_name: 'wp:object',
+										_attrs: {
+											xmlns: `http://wordpress.org/export/objects/${ objectType }/1.0/`,
```

Comment:
I am no big expert in XML but I wasn't able to read this from simplexml since `xmlns` is reserved and usually defines a prefix for the subtree, so we'd need something like `xmlns:o` and then prefix everything with an `o:`